extensions/amp-user-notification/0.1/amp-user-notification.js

```diff
@@ -0,0 +1,335 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Layout} from '../../../src/layout';
+import {assertHttpsUrl} from '../../../src/url';
+import {assert} from '../../../src/asserts';
+import {cidFor} from '../../../src/cid';
+import {getService} from '../../../src/service';
+import {log} from '../../../src/log';
+import {xhrFor} from '../../../src/xhr';
+
+
+const TAG_ = 'AmpUserNotification';
+
+/**
+ * @typedef {{
+ *   elementId: string,
+ *   ampUserId: string
+ * }}
+ */
+let GetRequestMetadata;
```

Comment:
correct. made changes and made sure to access value through bracket notation `data['showNotification']` also added `@export` to typedef so that they wont be mangled (correct me if im wrong on that last note).